### PR TITLE
Changes to vendor API docs for version change

### PR DESCRIPTION
**Pull request by SamJamCul**
## Context

https://trello.com/c/je9Cwcv3/5065-amend-draft-api-content

## Changes proposed in this pull request

Replaced the `Versioning` section with the `API Versioning Strategy` sections set out in the ticket doc
Replaced the `API Versioning Strategy` with a redirection line in the draft docs
Added the `Testing draft version 1.1` in the draft docs

## Guidance to review

head to `/api-docs` and `/api-docs/draft` to review the changes

### CODE REVIEW

File: app/views/api_docs/vendor_api_docs/pages/home.md [hunk 1]
```diff
@@ -75,21 +75,71 @@ Each token is associated with a single provider. It will grant access to applica
...
+
+### How the API version number reflects the changes we’ve made
+
+We use the format major.minor (for example, 1.2) to indicate the API version.
+
+The first number indicates a major version. This is incremented each time breaking changes are made, for example 1.2 changes to 2.0.
+
+The number after the decimal point indicates a minor version. This is incremented each time non-breaking changes are made, for example 1.2 changes to 1.3.
+
+The current version of this API is 1.0. The next version will be 1.1.
```
Comment (JR-G) [inline app/views/api_docs/vendor_api_docs/pages/home.md:115]: I guess one to remember for next week!

File: app/views/api_docs/vendor_api_docs/pages/home.md [hunk 2]
```diff
@@ -75,21 +75,71 @@ Each token is associated with a single provider. It will grant access to applica
...
+
+This means that if you use the major version URL, you do not need to update student record systems every time we make a minor update.
+
+For example, after version 1.1 is released you can use:
+
+- https://apply-for-teacher-training.service.gov.uk/api/v1.0 for version 1.0
+- https://apply-for-teacher-training.service.gov.uk/api/v1.1 for version 1.1
+- https://apply-for-teacher-training.service.gov.uk/api/v1 for version 1.1 - but if version 1.2 is released then this URL will give you version 1.2 instead
+
+## How applications are updated
```
Comment (JR-G) [inline app/views/api_docs/vendor_api_docs/pages/home.md:134]: Just noticed the formatting of this is slightly off on the review app:
image

Comment (SamJamCul) [reply app/views/api_docs/vendor_api_docs/pages/home.md:134]: Nice one, my IDE is formatting slightly differently to the frontend. It's been fixed now

File: app/views/api_docs/vendor_api_docs/pages/home.md (hunk 1 above)
Comment (SamJamCul) [reply app/views/api_docs/vendor_api_docs/pages/home.md:115]: Would be cool if we could join that to the current API version, so it's always reading `The current version of this API is API::VERSION. The next version will be API::VERSION+1`

File: app/views/api_docs/vendor_api_docs/pages/home.md [hunk 3]
```diff
@@ -75,21 +75,73 @@ Each token is associated with a single provider. It will grant access to applica
...
+
+We use the format major.minor (for example, 1.2) to indicate the API version.
+
+The first number indicates a major version. This is incremented each time breaking changes are made, for example 1.2 changes to 2.0.
+
+The number after the decimal point indicates a minor version. This is incremented each time non-breaking changes are made, for example 1.2 changes to 1.3.
+
+The current version of this API is 1.0. The next version will be 1.1.
+
+Changes are documented in our release notes.
```
Comment (cwrw) [inline app/views/api_docs/vendor_api_docs/pages/home.md:117]: could you link to the release notes page please?

File: app/views/api_docs/vendor_api_docs/pages/home.md [hunk 4]
```diff
@@ -75,21 +75,73 @@ Each token is associated with a single provider. It will grant access to applica
...
+- endpoints, for example to allow individual conditions to be marked as met or not met
+- nested resources and objects, for example details of interviews
+- fields, for example when candidates are asked a new question
+- optional query parameters, for example to allow optional pagination when applications are synced
+
+You do not need to update student record systems before moving to a new version with non-breaking changes. You only need to make updates if you want to use the version’s new features.
+
+### How the API version number reflects the changes we’ve made
+
+We use the format major.minor (for example, 1.2) to indicate the API version.
```
Comment (cwrw) [inline app/views/api_docs/vendor_api_docs/pages/home.md:109]: could we wrap version numbers with `` ?

File: app/views/api_docs/vendor_api_docs/pages/home.md [hunk 5]
```diff
@@ -75,21 +75,73 @@ Each token is associated with a single provider. It will grant access to applica
...
+When an API version is officially released, minor version updates will be made available:
+
+- on their own minor version URL, for example v1.1
+- on a major version URL which does not indicate a minor version, for example v1
+
+This means that if you use the major version URL, you do not need to update student record systems every time we make a minor update.
+
+For example, after version 1.1 is released you can use:
+
+- https://apply-for-teacher-training.service.gov.uk/api/v1.0 for version 1.0
```
Comment (cwrw) [inline app/views/api_docs/vendor_api_docs/pages/home.md:130]: could we format those as links please?

File: app/views/api_docs/vendor_api_docs/pages/home.md (hunk 4 above)
Comment (SamJamCul) [reply app/views/api_docs/vendor_api_docs/pages/home.md:109]: when we've got `v1.1` should the v be included in the \`\`

File: app/views/api_docs/vendor_api_docs/pages/home.md (hunk 5 above)
Comment (SamJamCul) [reply app/views/api_docs/vendor_api_docs/pages/home.md]: sorted

File: app/views/api_docs/vendor_api_docs/pages/home.md (hunk 3 above)
Comment (SamJamCul) [reply app/views/api_docs/vendor_api_docs/pages/home.md]: done

**Review by cwrw — APPROVED** [review]
thanks @SamJamCul

**Pull request merged**